palette/src/matrix.rs

```diff
@@ -52,48 +68,54 @@ pub fn multiply_rgb_to_xyz<S: RgbSpace, T: FloatComponent>(
 
 /// Multiply two 3x3 matrices.
 pub fn multiply_3x3<T: Float>(c: &Mat3<T>, f: &Mat3<T>) -> Mat3<T> {
-    let mut out = [T::zero(); 9];
-    out[0] = c[0] * f[0] + c[1] * f[3] + c[2] * f[6];
-    out[1] = c[0] * f[1] + c[1] * f[4] + c[2] * f[7];
-    out[2] = c[0] * f[2] + c[1] * f[5] + c[2] * f[8];
+    assert!(c.len() > 8);
+    assert!(f.len() > 8);
+
+    let o0 = c[0] * f[0] + c[1] * f[3] + c[2] * f[6];
+    let o1 = c[0] * f[1] + c[1] * f[4] + c[2] * f[7];
+    let o2 = c[0] * f[2] + c[1] * f[5] + c[2] * f[8];
 
-    out[3] = c[3] * f[0] + c[4] * f[3] + c[5] * f[6];
-    out[4] = c[3] * f[1] + c[4] * f[4] + c[5] * f[7];
-    out[5] = c[3] * f[2] + c[4] * f[5] + c[5] * f[8];
+    let o3 = c[3] * f[0] + c[4] * f[3] + c[5] * f[6];
+    let o4 = c[3] * f[1] + c[4] * f[4] + c[5] * f[7];
+    let o5 = c[3] * f[2] + c[4] * f[5] + c[5] * f[8];
 
-    out[6] = c[6] * f[0] + c[7] * f[3] + c[8] * f[6];
-    out[7] = c[6] * f[1] + c[7] * f[4] + c[8] * f[7];
-    out[8] = c[6] * f[2] + c[7] * f[5] + c[8] * f[8];
+    let o6 = c[6] * f[0] + c[7] * f[3] + c[8] * f[6];
+    let o7 = c[6] * f[1] + c[7] * f[4] + c[8] * f[7];
+    let o8 = c[6] * f[2] + c[7] * f[5] + c[8] * f[8];
 
-    out
+    [o0, o1, o2, o3, o4, o5, o6, o7, o8]
```

Comment:
The destructuring seemed to help here. There's no need to zero-out an array since we never do anything with that value.

Reply:
I think it's a nice stylistic improvement too. 👍